datashuttle/tui/interface.py

```diff
@@ -224,11 +224,28 @@ def transfer_top_level_only(
             from central to remote.
 
         """
+        assert selected_top_level_folder in ["rawdata", "derivatives"]
+
         try:
-            if upload:
-                self.project.upload_all(selected_top_level_folder)
+            if selected_top_level_folder == "rawdata":
+                transfer_func = (
+                    self.project.upload_rawdata
+                    if upload
+                    else self.project.download_rawdata
+                )
+            elif selected_top_level_folder == "derivatives":
+                transfer_func = (
+                    self.project.upload_derivatives
+                    if upload
+                    else self.project.download_derivatives
+                )
             else:
-                self.project.download_all(selected_top_level_folder)
+                raise ValueError(
+                    "`selected_top_level_folder` must be 'rawdata' or 'derivatives'."
+                )
```

Comment:
Since you are raising this error here, what's the use of the earlier assertion?

Reply:
There is none! must have missed it cheers